src/divider/divider.style.tsx

```diff
@@ -0,0 +1,64 @@
+import styled, { css } from "styled-components";
+import { Color } from "../color/color";
+import { Layout } from "../layout";
+import { DividerLineStyleType } from "./types";
+
+// =============================================================================
+// STYLE INTERFACES
+// =============================================================================
+interface StyleProps {
+    $thickness: number;
+    $color?: string | ((props: any) => string);
+    $lineStyle?: DividerLineStyleType;
+}
+
+// =============================================================================
+// STYLING
+// =============================================================================
+export const Line = styled.hr<StyleProps>`
+    width: 100%;
+    background-color: transparent;
+    margin: 0;
+    height: unset;
+    border: none;
+    ${(props) => {
+        return css`
+            border-top-width: ${props.$thickness}px;
+            border-top-style: ${props.$lineStyle} !important; // Prevent modification easily
+            border-top-color: ${props.$color || Color.Neutral[5]};
+        `;
+    }}
+`;
+
+const dashedLineStyle = (props: StyleProps) => {
+    let color;
+
+    if (props.$color && typeof props.$color === "function") {
+        color = props.$color(props);
+    } else {
+        color = props.$color || Color.Neutral[5](props);
+    }
+
+    const encodedColor = encodeURIComponent(color);
+    const thickness = props.$thickness || 1;
+    const strokeWidth = thickness + 1; // best fit
+
+    return css`
+        display: block;
+        width: 100%;
+        background-color: transparent;
+        margin: 0;
+        height: ${thickness}px;
+        border: none;
+        background-repeat: repeat-x;
+        background-image: url('data:image/svg+xml,<svg width="8" height="${thickness}" viewBox="0 0 8 1" xmlns="http://www.w3.org/2000/svg"><line x1="2" y1="1" x2="8" y2="1" stroke="${encodedColor}" stroke-width="${strokeWidth}" stroke-dasharray="4 4" /></svg>');
```

Comment:
instead of using different components for each line style, could this `styled.hr` and svg definition be reused? then I'm thinking the col div can just be conditionally rendered based on whether its layout is grid

Reply:
No longer relevant as the components have been merged into a single styled component